rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -436,6 +441,26 @@ def _parse_slot(self, step: Dict[Text, Any]) -> None:
                 )
                 return
 
+    @functools.lru_cache()
+    def _slot_default_value(self, slot_name: Text) -> Any:
+        if not self.domain:
+            return None
+
+        slot_types_with_default_types = {TextSlot: "filled", ListSlot: ["filled"]}
+        slot = next(slot for slot in self.domain.slots if slot.name == slot_name)
+
+        default_value = slot_types_with_default_types.get(type(slot))
+        if default_value is None and slot.has_features():
+            rasa.shared.utils.io.raise_warning(
+                f"Slot '{slot_name}' was referenced by its name only. As slot "
+                f"'{slot_name}' is of type '{slot.type_name}' you need to specify a "
+                f"value for it. Slot '{slot_name}' will be treated as if it's value "
+                f"is empty.",
+                docs=DOCS_URL_SLOTS,
+            )
```

Comment:
Should this be an error instead of a warning?

Reply:
All the other stuff is warnings, that's why I chose `raise_warning` to be consistent.

Reply:
👍